packages/flutter_tools/test/general.shard/resident_runner_test.dart

```diff
@@ -336,21 +337,32 @@ void main() {
 
     // supports service protocol
     expect(residentRunner.supportsServiceProtocol, true);
-    expect(bufferLogger.statusText, contains('"w"'));
-    expect(bufferLogger.statusText, contains('"t"'));
-    expect(bufferLogger.statusText, contains('"P"'));
-    expect(bufferLogger.statusText, contains('"a"'));
     // isRunningDebug
     expect(residentRunner.isRunningDebug, true);
-    expect(bufferLogger.statusText, contains('"L"'));
-    expect(bufferLogger.statusText, contains('"S"'));
-    expect(bufferLogger.statusText, contains('"U"'));
-    expect(bufferLogger.statusText, contains('"i"'));
-    expect(bufferLogger.statusText, contains('"p"'));
-    expect(bufferLogger.statusText, contains('"o"'));
-    expect(bufferLogger.statusText, contains('"z"'));
-    // screenshot
-    expect(bufferLogger.statusText, contains('"s"'));
+    // commands
+    expect(bufferLogger.statusText, equals(
+        <dynamic>[
+          'Flutter run key commands.',
+          CommandHelp.r(),
+          CommandHelp.R(),
+          CommandHelp.h(),
+          CommandHelp.q(),
+          CommandHelp.s(),
+          CommandHelp.w(),
+          CommandHelp.t(),
+          CommandHelp.L(),
+          CommandHelp.S(),
+          CommandHelp.U(),
+          CommandHelp.i(),
+          CommandHelp.p(),
+          CommandHelp.o(),
+          CommandHelp.z(),
+          CommandHelp.P(),
+          CommandHelp.a(),
```

Comment:
expand these out to the actual string, that way you get to test the CommandHelp logic as well essentially for free.

Reply:
This is done automatically in `join` so we do touch the `CommandHelp.toString` on all the commands.

Reply:
I mean that you don't check what the output actually is, you just check that it matches what CommandHelp returns. But we know it returns that, because that's what the code does. What if someone changes the actual help text accidentally? Nothing would catch that.

Reply:
OK, I see what you are saying. We should safeguard the help text.